lib/debug/debugger.dart

```diff
@@ -160,8 +163,26 @@ class UriResolver implements Disposable {
     if (_uriToPath.containsKey(uri)) return _uriToPath[uri];
 
     String contextId = await _completer.future;
-    MapUriResult result = await analysisServer.server.execution.mapUri(contextId, uri: uri);
-    String path = result.file;
+    String path;
+    try {
```

Comment:
Hmm, I'm curious what this is doing. You're using the analysis server to resolve a uri to a file path? There is an API for that, in the debugger domain.

I may not understand what this is trying to do though - I'd suggest adding some descriptive comments :)

Reply:
It is at least very creative :)

Reply:
:)  Added comment.  If you think it is an analyzer bug, please let me know.